pandas/core/internals/blocks.py

```diff
@@ -1950,6 +1973,24 @@ def shift(self, periods, axis=0):
     def _ftype(self):
         return getattr(self.values, '_pandas_ftype', Block._ftype)
 
+    def _unstack(self, unstacker_func, new_columns, n_rows, fill_value):
+        dummy_arr = np.arange(n_rows)
+        dummy_unstacker = functools.partial(unstacker_func, fill_value=-1)
+        unstacker = dummy_unstacker(dummy_arr)
+
+        new_placement, new_values, mask = self._get_unstack_items(
+            unstacker, new_columns
```

Comment:
would not this generically work for all unstacking/ (e.g. what if you make this the super method)?

Reply:
I think this is slower in general. It's necessarily slower for NumPy types, since you have to do the reshaping / unstack on the ndarray of positions anyway.

The hope is that the cost of the additional `take` per column is offset by not converting to an ndarray[object] in the first place.